project_task_pull_request_state/models/project_task.py

```diff
@@ -0,0 +1,39 @@
+from odoo import fields, models, api
+
+
+class ProjectTaskState(models.Model):
+    _inherit = "project.task"
+
+    pr_state = fields.Selection(
+        selection=lambda self: self._selection_pr_state(),
+        string="PR State",
+        )
+
+    @api.model
+    def _selection_pr_state(self):
+        """Function to select the state of the pull request"""
+        return [
+            ('open', 'Open'),
+            ('draft', 'Draft'),
+            ('merged', 'Merged'),
+            ('closed', 'Closed'),
+        ]
+
+    @api.model
+    def create(self, vals):
+        self._check_pr_state(vals)
```

Comment:
_check_pr_state() function requires record set in self.
However there is are no records at this point because nothing is created yet